File: 2021/02/Part1.cpp
#include <string>
#include <vector>

#include "lib.h"

int main() {
    std::vector<std::string> lines = readLines("input.txt");

    int depth = 0;
    int horpos = 0;

    for (std::string line: lines) {
        std::vector<std::string> splitLine = split(line, " ");
        if (splitLine.size() != 2) {
            puts("Incorrect size");
            return -1;
        }
        std::string dir = splitLine.at(0);
        int x = std::stoi(splitLine.at(1));
        if (dir.compare("forward") == 0) {
            horpos += x;
        } else if (dir.compare("down") == 0) {
            depth += x;
        } else if (dir.compare("up") == 0) {
            depth -= x;
        }
    }

    printf("%d\n", depth * horpos);

    return 0;
}

File: 2022/04/Part1.py
t=0
with open("input.txt") as f:
    for line in f:
        p1,p2=line.split(",")
        p11,p12=p1.split("-")
        p21,p22=p2.split("-")
        p11, p12, p21, p22 = int(p11),int(p12),int(p21),int(p22)
        if (p11 <= p21 and p22 <= p12) or (p21 <= p11 and p12 <= p22):
           t+=1
print(t)